File: react/src/api/models/acroStack/services/dtos/cms/UpdateBlogPostInput.ts
/* oxlint-disable */

/**
 * @type object
 */
export type AcroStackServicesDtosCmsUpdateBlogPostInput = {
  /**
   * @minLength 0
   * @maxLength 256
   * @type string
   */
  title: string;
  /**
   * @minLength 0
   * @maxLength 256
   * @type string
   */
  slug: string;
  /**
   * @minLength 1
   * @type string
   */
  content: string;
  /**
   * @minLength 0
   * @maxLength 512
   * @type string | undefined
   */
  excerpt?: string | null;
  /**
   * @minLength 0
   * @maxLength 512
   * @type string | undefined
   */
  coverImage?: string | null;
  /**
   * @type array | undefined
   */
  tags?: string[] | null;
};

File: AcroStack/Services/Dtos/Cms/UpdateBlogPostInput.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace AcroStack.Services.Dtos.Cms;

/// <summary>Input for updating an existing <see cref="BlogPostDto"/>.</summary>
public class UpdateBlogPostInput
{
    [Required]
    [StringLength(256)]
    public string Title { get; set; } = string.Empty;

    [Required]
    [StringLength(256)]
    public string Slug { get; set; } = string.Empty;

    [Required]
    public string Content { get; set; } = string.Empty;

    [StringLength(512)]
    public string? Excerpt { get; set; }

    [StringLength(512)]
    public string? CoverImage { get; set; }

    /// <summary>Tag names to associate with the post (replaces existing tags).</summary>
    public List<string> Tags { get; set; } = new();
}
